source/extensions/filters/http/decompressor/decompressor_filter.cc

```diff
@@ -82,9 +87,21 @@ Http::FilterHeadersStatus DecompressorFilter::decodeHeaders(Http::RequestHeaderM
                              *decoder_callbacks_, headers);
 };
 
-Http::FilterDataStatus DecompressorFilter::decodeData(Buffer::Instance& data, bool) {
-  return maybeDecompress(config_->requestDirectionConfig(), request_decompressor_,
-                         *decoder_callbacks_, data);
+Http::FilterDataStatus DecompressorFilter::decodeData(Buffer::Instance& data, bool end_stream) {
+  if (request_decompressor_) {
+    HeaderMapOptRef trailers;
+    if (end_stream) {
+      trailers = HeaderMapOptRef(std::ref(decoder_callbacks_->addDecodedTrailers()));
```

Comment:
I assume `addDecodedTrailers()` returns the current trailers if another filter has already added them?

Reply:
if any other filter (or the request itself) added/had trailers we would not get a decodeData with `end_stream==True`.